Added Section on Access Modifiers (#65)
BHoM/documentation: docs/Contributing/Development FAQ/Coding-Style.md

- to ensure the functionality can be correctly exposed to the UIs;
- to organise functionality and classes in a tidy, easy-to-find manner.

### Access Modifiers

> Access modifiers specify the accessibility level of type and type members. They denote whether if a type or member can be used from other code in the same assembly, and in other assemblies.

Reviewer: ```suggestion
Access modifiers specify the accessibility level of type and type members. They denote whether a type or member can be used by other code in the same assembly, and in other assemblies.
```

> - In line with BHoM's focus on claity and trasparency, we generally use the `public` access modifier, which allows a type or member to be accessed by any other code in the same assembly or other assembly that refernces it.

Reviewer: ```suggestion
- In line with BHoM's focus on clarity and transparency, we generally use the `public` access modifier, which allows a type or member to be accessed by any other code in the same assembly or other assembly that reference it.
```
